### PR TITLE
[OCDM] Enhancements

### PR DESCRIPTION
**Pull request by sramani-metro**
1. Support passing Encryption Scheme and Pattern to Decrypt
    2. New Gstreamer Adapter method that parses Sample and Subsample info
       from GstStructure in ProitectionMetdata of GstBuffer
    3. New gstreamer adapter API supports passing Subsample Info to the DRM
       implementation without processing it.

### CODE REVIEW

File: Source/ocdm/open_cdm_impl.h [hunk 1]
```diff
@@ -2,7 +2,7 @@
  * If not stated otherwise in this file or this component's LICENSE file the
  * following copyright and licenses apply:
  *
- * Copyright 2020 RDK Management
+ * Copyright 2020 Metrological
```
Comment (anand-ky) [inline Source/ocdm/open_cdm_impl.h:5]: Pls restore Copyright to RDK Management.

File: Source/ocdm/open_cdm_ext.cpp [hunk 2]
```diff
@@ -2,7 +2,7 @@
  * If not stated otherwise in this file or this component's LICENSE file the
  * following copyright and licenses apply:
  *
- * Copyright 2020 RDK Management
+ * Copyright 2020 Metrological
```
Comment (anand-ky) [inline Source/ocdm/open_cdm_ext.cpp:5]: Pls restore Copyright to RDK Management.

File: Source/ocdm/Module.h [hunk 3]
```diff
@@ -2,7 +2,7 @@
  * If not stated otherwise in this file or this component's LICENSE file the
  * following copyright and licenses apply:
  *
- * Copyright 2020 RDK Management
+ * Copyright 2020 Metrological
```
Comment (anand-ky) [inline Source/ocdm/Module.h:5]: Pls restore Copyright to RDK Management.

File: Source/ocdm/adapter/gstreamer/open_cdm_adapter.cpp [hunk 4]
```diff
@@ -2,7 +2,7 @@
  * If not stated otherwise in this file or this component's LICENSE file the
  * following copyright and licenses apply:
  *
- * Copyright 2020 RDK Management
+ * Copyright 2020 Metrological
```
Comment (anand-ky) [inline Source/ocdm/adapter/gstreamer/open_cdm_adapter.cpp:5]: Pls restore Copyright to RDK Management.

**Review by anand-ky — CHANGES_REQUESTED** [review]

File: Source/ocdm/open_cdm_impl.h (hunk 1 above)
Comment (sramani-metro) [reply Source/ocdm/open_cdm_impl.h]: Done

File: Source/ocdm/open_cdm_ext.cpp (hunk 2 above)
Comment (sramani-metro) [reply Source/ocdm/open_cdm_ext.cpp]: Done

File: Source/ocdm/adapter/gstreamer/open_cdm_adapter.cpp (hunk 4 above)
Comment (sramani-metro) [reply Source/ocdm/adapter/gstreamer/open_cdm_adapter.cpp]: Done

File: Source/ocdm/Module.h (hunk 3 above)
Comment (sramani-metro) [reply Source/ocdm/Module.h]: Done

**Review by anand-ky — APPROVED** [review]

**Review by anand-ky — APPROVED** [review]

**Review by anand-ky — APPROVED** [review]

**Review by karuna2git — APPROVED** [review]

**Pull request merged**